build.gradle.kts

```
@@ -422,9 +424,86 @@ subprojects {
         }
     }
 
+    plugins.withId("me.champeau.gradle.japicmp") {
+        afterEvaluate {
+            // hack to find the current released version of the project
+            val temp: Configuration = project.configurations.create("tempConfig")
+            project.dependencies.add("tempConfig", "io.opentelemetry:opentelemetry-bom:latest.release")
+            val moduleVersion = project.configurations["tempConfig"].resolvedConfiguration.firstLevelModuleDependencies.elementAt(0).moduleVersion
+            project.configurations.remove(temp)
+
+            val userRequestedBase = project.properties["apiBaseVersion"]
+            val baselineVersion = userRequestedBase ?: moduleVersion
```

```suggestion
            val baselineVersion = userRequestedBase ?: "latest.release"
```

And don't need the previous logic, I think?

---

I'll try it. not sure if it'll properly resolve or not

---

hmm. the issue is that we don't have a way to name the jar file down on these lines:
``` 
                val depModule = "io.opentelemetry:${base.archivesBaseName}:$baselineVersion@jar"
                val depJar = "${base.archivesBaseName}-${baselineVersion}.jar"
```
There may be another way to do it, but we'd probably have to start hacking on the gradle plugin (which I want to do to fix some bugs anyway)